static/js/formatters-internal.js

```diff
@@ -348,22 +376,80 @@ export function image(simpleOrComplexImage = {}, desiredSize = '200x', atLeastAs
 
   const urlWithDesiredDims = urlBeforeDimensions + desiredWidth + 'x' + desiredHeight + extension;
 
-  const dynamicUrl = atLeastAsLarge
+  return atLeastAsLarge
     ? _replaceUrlHost(urlWithDesiredDims, 'dynl.mktgcdn.com')
     : _replaceUrlHost(urlWithDesiredDims, 'dynm.mktgcdn.com');
+}
 
-  return Object.assign(
-    {},
-    image,
-    {
-      url: dynamicUrl.replace('http://', 'https://')
+/**
+ * Given an EU image url, returns the dynamic url.
+ *
+ * @param {Object} image The image object. (e.g.
+ * {
+ *   url: 'https://a.eu.mktgcdn.com/f/0/FLVfkpR1IwpWrWDuyNYCJWVYIDfPO6x1QSztXozMIzo.jpg',
+ *   sourceUrl: 'https://a.mktgcdn.com/p/UN9RPhz0V9D8bNZ3XfNpkGnAk6ikFhVmgvntlBjVyMA/1200x675.jpg',
+ *   width: 1200,
+ *   height: 675,
+ * })
+ * @param {string} desiredSize The desired size of the image ('<width>x<height>')
+ * @param {boolean} atLeastAsLarge Whether the image should be at least as large as the desired
+ *                                 size in one dimension or smaller than the desired size in both
+ *                                 dimensions.
+ * @returns {string} A dynamic url (e.g. 'https://dyn.eu.mktgcdn.com/f/0/FLVfkpR1IwpWrWDuyNYCJWVYIDfPO6x1QSztXozMIzo.jpg/width=200,fit=contain')
+ */
+function _getEuImageDynamicUrl(image, desiredSize, atLeastAsLarge) {
+  let fullSizeWidth, fullSizeHeight;
+  if (image.width) {
+    fullSizeWidth = image.width;
+  }
+  if (image.height) {
+    fullSizeHeight = image.height;
+  }
+
+  if (image.sourceUrl && (!fullSizeWidth || !fullSizeHeight)) {
+    const [urlWithoutExtension, _] = _splitStringOnIndex(image.sourceUrl, image.sourceUrl.lastIndexOf('.'));
+    const [__, dimensions] = _splitStringOnIndex(urlWithoutExtension, urlWithoutExtension.lastIndexOf('/') + 1);
+    const fullSizeDims = dimensions.split('x');
+
+    fullSizeWidth = Number.parseInt(fullSizeDims[0]);
+    fullSizeHeight = Number.parseInt(fullSizeDims[1]);
+  }
+
+  let desiredDims = desiredSize.split('x');
+  let formatOptions = [];
+
+  if (desiredDims[0] !== '') {
+    const desiredWidth = Number.parseInt(desiredDims[0]);
+    if (Number.isNaN(desiredWidth)) {
+      throw new Error("Invalid width specified");
     }
-  );
+
+    formatOptions.push(`width=${desiredWidth}`);
+  } else if (!atLeastAsLarge && fullSizeWidth) {
```

Comment:
to double check, we don't want to add the `width` param unless the image should be contained by that width?

Reply:
yep. I was trying to match what we are doing for US images based on my understanding.

Reply:
but I think the US images add a size of 1 for any missing dimension when `atLeastAsLarge=true`

Reply:
actually, I think the current behavior is correct for EU. looks like `dynm`/`dynl` end up ignoring the dimensions with size 1 for US images anyways ([thread](https://yext.slack.com/archives/C01AFQG2ASH/p1633797078031300?thread_ts=1633115673.019900&cid=C01AFQG2ASH))